src/components/Swap/index.tsx

```diff
@@ -62,6 +63,7 @@ export default function Swap(props: SwapProps) {
   useSyncSwapEventHandlers(props as SwapEventHandlers)
   useSyncTokenDefaults(props as TokenDefaults)
   useSyncWidgetEventHandlers(props as WidgetEventHandlers)
+  useSyncBrandingSetting(props as BrandingSetting)
```

Comment:
This should be sync'ed as part of the `Widget` component, not as part of the `Swap` component, as it is widget-scoped state.

Reply:
I had the same thought, but the AtomProvider is at the `Widget` component level, so only its children can share atoms

Reply:
This will need to be moved eventually (with `useSyncWidgetEventHandlers`) but I think it's fine for now.

Reply:
On L53, `SwapProps` must extend `BrandingSettings` so that the type system will enforce that these props are available.